src/components/state-history-chart-line.html

```diff
@@ -1,265 +1,269 @@
+      return;
     }
 
-    connectedCallback() {
-      super.connectedCallback();
-      this._isAttached = true;
-      this.drawChart();
-      this.addEventListener('iron-resize', () => {
-        setTimeout(() => this.drawChart(), 10);
-      });
+    function safeParseFloat(value) {
+      var parsed = parseFloat(value);
+      // isFinite(NaN) === False
+      return isFinite(parsed) ? parsed : null;
     }
 
-    dataChanged() {
-      this.drawChart();
+    // startTime = new Date(Math.min.apply(null, deviceStates.map(function (states) {
+    //   return new Date(states.states[0].last_changed);
+    // })));
+
+    endTime = this.endTime ||
+      new Date(Math.max.apply(null, deviceStates.map(states =>
+        new Date(states.states[states.states.length - 1].last_changed))));
+    if (endTime > new Date()) {
+      endTime = new Date();
     }
 
-    drawChart() {
-      var unit = this.unit;
-      var deviceStates = this.data;
-      var options;
-      var startTime;
-      var endTime;
-      var dataTables;
-      var finalDataTable;
-      var daysDelta;
+    datasets = [];
 
-      if (!this._isAttached) {
-        return;
-      }
+    deviceStates.forEach(function (states) {
+      var domain = states.domain;
+      var name = states.name;
+      // array containing [value1, value2, etc]
+      var prevValues;
+      var hasTargetRange;
+      var processState;
+      var data = [];
 
-      if (!this.chartEngine) {
-        this.chartEngine = new window.google.visualization.ComboChart(this);
-      }
 
-      if (deviceStates.length === 0) {
-        return;
+      function pushData(timestamp, datavalues) {
+        if (!datavalues) return;
+        if (timestamp > endTime) {
+          // Drop datapoints that are after the requested endTime. This could happen if
+          // endTime is "now" and client time is not in sync with server time.
+          return;
+        }
+        data.forEach(function (d, i) {
+          d.data.push({ x: timestamp, y: datavalues[i] });
+        });
+        prevValues = datavalues;
       }
 
-      options = {
-        backgroundColor: '#fafafa',
-        legend: { position: 'top' },
-        interpolateNulls: true,
-        titlePosition: 'none',
-        vAxes: {
-          // Adds units to the left hand side of the graph
-          0: { title: unit },
-        },
-        hAxis: {
-          format: 'H:mm',
-        },
-        chartArea: { left: '60', width: '95%' },
-        explorer: {
-          actions: ['dragToZoom', 'rightClickToReset', 'dragToPan'],
-          keepInBounds: true,
-          axis: 'horizontal',
-          maxZoomIn: 0.1,
-        },
-        seriesType: 'line',
-      };
-
-      if (this.isSingleDevice) {
-        options.legend.position = 'none';
-        options.vAxes[0].title = null;
-        options.chartArea.left = 40;
-        options.chartArea.height = '80%';
-        options.chartArea.top = 5;
+      function addColumn(nameY, type) {
```

Comment:
I think there is no need to do string manipulations - just pass fill and step as params.